src/main/java/net/minecraftforge/items/ItemHandlerHelper.java

```diff
@@ -0,0 +1,38 @@
+package net.minecraftforge.items;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+
+public class ItemHandlerHelper {
+    public static ItemStack insertItem(IItemHandler dest, ItemStack stack, boolean simulate)
+    {
+        if (dest == null || stack == null)
+            return stack;
+
+        for (int i = 0; i < dest.getSlots() && stack != null; i++)
+        {
+            stack = dest.insertItem(i, stack, simulate);
+        }
+
+        return stack;
+    }
+
+    public static boolean canItemStacksStack(ItemStack a, ItemStack b)
+    {
+        if (a == null || !a.isItemEqual(b))
+            return false;
+
+        final NBTTagCompound aTag = a.getTagCompound();
+        final NBTTagCompound bTag = b.getTagCompound();
+        return (aTag != null || bTag == null) && (aTag == null || aTag.equals(bTag));
```

Comment:
Why not just
```aTag != null & aTag.equals(bTag)```
is there a point in checking  aTag.equals(bTag) if bTag is already null?